helix-view/src/document.rs

```diff
@@ -1080,20 +1080,26 @@ impl Document {
     }
 
     pub fn pickup_last_saved_time(&mut self) {
-        self.last_saved_time = match self.path.as_mut().unwrap().metadata() {
-            Ok(metadata) => match metadata.modified() {
-                Ok(mtime) => mtime,
-                Err(_) => {
+        self.last_saved_time = match self.path().as_mut() {
+            Some(path) => match path.metadata() {
+                Ok(metadata) => match metadata.modified() {
+                    Ok(mtime) => mtime,
+                    Err(_) => {
+                        log::error!(
+                            "Use a system time instead of fs' mtime not supported on this platform"
+                        );
+                        SystemTime::now()
+                    }
+                },
+                Err(e) => {
                     log::error!(
-                        "Use a system time instead of fs' mtime not supported on this platform"
+                        "Use a system time instead of fs' mtime: failed to file's metadata: {e}"
                     );
                     SystemTime::now()
                 }
             },
-            Err(e) => {
-                log::error!(
-                    "Use a system time instead of fs' mtime: failed to file's metadata: {e}"
-                );
+            None => {
+                log::error!("No current path, can't resolve last saved time");
```

Comment:
I'm not sure this log message is needed, maybe it could be removed completely?

Reply:
A log message is not needed, a scratch buffer is completely valid case

Reply:
Removed the log message